pkg/datagatherer/k8s/fieldfilter.go

```diff
@@ -1,143 +1,88 @@
 package k8s
 
 import (
-	"fmt"
-	"strings"
-
-	"github.com/Jeffail/gabs/v2"
-	json "github.com/json-iterator/go"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
 // SecretSelectedFields is the list of fields sent from Secret objects to the
 // backend
-var SecretSelectedFields = []string{
-	"kind",
-	"apiVersion",
-	"metadata.annotations",
-	"metadata.labels",
-	"metadata.name",
-	"metadata.namespace",
-	"metadata.ownerReferences",
-	"metadata.selfLink",
-	"metadata.uid",
-	"type",
-	"/data/tls.crt",
-	"/data/ca.crt",
+var SecretSelectedFields = []FieldPath{
+	{"kind"},
+	{"apiVersion"},
+	{"metadata", "annotations"},
+	{"metadata", "labels"},
+	{"metadata", "name"},
+	{"metadata", "namespace"},
+	{"metadata", "ownerReferences"},
+	{"metadata", "selfLink"},
+	{"metadata", "uid"},
+
+	{"type"},
+	{"data", "tls.crt"},
+	{"data", "ca.crt"},
 }
 
 // RouteSelectedFields is the list of fields sent from OpenShift Route objects to the
 // backend
-var RouteSelectedFields = []string{
-	"kind",
-	"apiVersion",
-	"metadata.annotations",
-	"metadata.name",
-	"metadata.namespace",
-	"metadata.ownerReferences",
-	"metadata.selfLink",
-	"metadata.uid",
-	"spec.host",
-	"spec.to.kind",
-	"spec.to.port",
-	"spec.to.name",
-	"spec.to.weight",
-	"spec.tls.termination",
-	"spec.tls.certificate",
-	"spec.tls.caCertificate",
-	"spec.tls.destinationCACertificate",
-	"spec.tls.insecureEdgeTerminationPolicy",
-	"spec.wildcardPolicy",
-	"status",
+var RouteSelectedFields = []FieldPath{
+	{"kind"},
+	{"apiVersion"},
+	{"metadata", "annotations"},
+	{"metadata", "name"},
+	{"metadata", "namespace"},
+	{"metadata", "ownerReferences"},
+	{"metadata", "selfLink"},
+	{"metadata", "uid"},
+
+	{"spec", "host"},
+	{"spec", "to", "kind"},
+	{"spec", "to", "name"},
+	{"spec", "to", "weight"},
+	{"spec", "tls", "termination"},
+	{"spec", "tls", "certificate"},
+	{"spec", "tls", "caCertificate"},
+	{"spec", "tls", "destinationCACertificate"},
+	{"spec", "tls", "insecureEdgeTerminationPolicy"},
+	{"spec", "wildcardPolicy"},
+	{"status"},
 }
 
 // RedactFields are removed from all objects
-var RedactFields = []string{
-	"metadata.managedFields",
-	"/metadata/annotations/kubectl.kubernetes.io~1last-applied-configuration",
```

Comment:
Oh, maybe something to do with handling field path segments that may contain `/`.